### PR TITLE
Remove setup.py and move to hatch build

### PR DESCRIPTION
**Pull request by rosteen**
I confirmed that this fixes the problem we were seeing with local editable installs in more recent versions of python. I haven't yet confirmed that it will fix the same problem (missing executables) when doing `pip install hstaxe` in 3.10, and although I strongly think it will, I'm not sure of the best way to test that before actually doing a release. Perhaps we merge this, do a release, and then we can do a hotfix release if it doesn't work (or breaks things further). Not exactly ideal to be sure.

### CODE REVIEW

**Comment by rosteen** [comment]
Whoops, meant to open this as draft, since I don't actually know yet how to trigger the "clean" build command as currently configured - I need to do a little more reading.

**Comment by rosteen** [comment]
I'm moving this back to ready for review, Duy agreed that it's not a big deal to have this `clean` function readily available when users can just remove their `hstaxe` env if desired.

**Comment by rosteen** [comment]
I'm moving this back to ready for review, Duy agreed that it's not a big deal to have this `clean` function readily available when users can just remove their `hstaxe` env if desired.

**Comment by ofek** [comment]
Awesome stuff! Is cleaning not working?

fyi the site's html looks like maybe a build failed https://www.stsci.edu

**Comment by rosteen** [comment]
@ofek It's not that the `clean` functionality isn't working, necessarily, the issue is that I haven't figured out how to actually trigger it in this `hatch` paradigm in an analogous manner to doing `python setup.py clean`.

**Comment by ofek** [comment]
Just `hatch clean` should do it

**Comment by rosteen** [comment]
I now have the `pip install` working on python 3.10 - you can test via test PyPi with the command `pip install --extra-index-url https://test.pypi.org/simple/ hstaxe==1.0.2RC1`.

@ofek Thanks for the tip on `hatch clean`! I ran into one problem that I haven't dug into yet, which is that we try to remove some executable binaries from the conda environment bin path, but `hatch clean` looks for them instead in the temporary hatch venv. This isn't a blocker for this and I'll probably read through the docs soon to see if I can resolve it, but if you happen to know the answer off the top of your head I definitely wouldn't mind 😁 . Thanks again for checking in here!

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,73 @@
 [build-system]
-requires = ["setuptools>=42", "setuptools_scm[toml]>=3.4", "wheel", "numpy"]
-build-backend = "setuptools.build_meta"
+requires = ["hatchling", "hatch-vcs"]
```
Comment (pllim) [inline pyproject.toml:2]: Would it be good idea to pin versions here so new hatch release won't accidentally break the build?

**Comment by pllim** [comment]
p.s. Are you not going to generate wheels?

**Comment by rosteen** [comment]
> p.s. Are you not going to generate wheels?

Probably not - there are no wheels currently, we would have to (I believe) create OS-specific wheels, and this package is pretty much in minimal-effort support mode so I don't want to put too much time into it. That said, I was able to build a wheel successfully with this PR, so perhaps we should.

File: pyproject.toml (hunk 1 above)
Comment (rosteen) [reply pyproject.toml:2]: Probably! I'll do that shortly.

**Comment by pllim** [comment]
If you have C, probably needs OS specific ones, yeah...

**Comment by bjkuhn** [comment]
I just successfully built `hstaXe` using python 3.10. I installed `wcstools` from `astroconda` rather than `condo-forge` and for the `pip install` I followed: `pip install --extra-index-url https://test.pypi.org/simple/ hstaxe==1.0.2RC1`

**Comment by pllim** [comment]
Since when astroconda has support for Python 3.10? 👀

**Comment by larrybradley** [comment]
@pllim It doesn't.  `wcstools` is a pure-C package (no Python dependencies).

**Comment by pllim** [comment]
@jhunkeler , what is the wcstools in astroconda doing that conda-forge recipe isn't?

**Comment by larrybradley** [comment]
@pllim The `conda-forge` files/paths were renamed so as not to conflict with `wcslib`.  See https://github.com/conda-forge/staged-recipes/pull/20731#issuecomment-1284651771

**Comment by pllim** [comment]
There must be a way for hstaxe to accomodate for these new paths? 🤔

**Comment by rosteen** [comment]
> There must be a way for hstaxe to accommodate for these new paths? 🤔

I'm working on it, figuring out the right way/place to add these paths in `configure.ac` (I think that's the right place to do it).

**Comment by rosteen** [comment]
Also, the `wcstools` problem is independent of this PR and shouldn't block merging this.

**Review by larrybradley — COMMENTED** [review]
LGTM.  Are you still going to pin the `hatch` versions?

**Comment by pllim** [comment]
Sounds like you hatched a good plan. 😏

**Review by larrybradley — APPROVED** [review]

**Comment by ofek** [comment]
fyi the site's html looks like maybe a build failed https://www.stsci.edu

**Comment by rosteen** [comment]
> fyi the site's html looks like maybe a build failed https://www.stsci.edu

I forgot to say yesterday, I brought this to the attention of our web team and it seems to be fixed now. Thanks for pointing it out.

**Pull request merged**